dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/AbstractCrudController.java

```diff
@@ -449,6 +449,9 @@ protected final WebMessage postObject( ObjectReport objectReport )
         {
             webMessage.setHttpStatus( HttpStatus.CREATED );
             webMessage.setLocation( getSchema().getRelativeApiEndpoint() + "/" + objectReport.getUid() );
+
+            removeCacheEntriesForThisEntityAndUser();
```

Comment:
I think this is not ideal.. Hooking directly into AbstractCrudController is not save, metadata might be coming into the metadata importer/endpoint as well. I think it will be better to avoid caching this stuff in the first place instead of trying to keep the cache fresh like this.

Reply:
Cache removed